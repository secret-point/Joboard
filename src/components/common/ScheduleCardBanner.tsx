import React from 'react';
import { Row } from "@amzn/stencil-react-components/layout";

interface ScheduleCardBannerProps {

}

const ScheduleCardBanner = ( props: ScheduleCardBannerProps ) => {

    return (
        <Row className="scheduleCardBanner">
            <div>Bonus: USD30</div>
        </Row>
    )
}

export default ScheduleCardBanner;
