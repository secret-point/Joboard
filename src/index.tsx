import React from "react";
import ReactDOM from "react-dom";
import "./styles/index.css";
import App from "./App.container";
import store from "./store";
import { Provider } from "react-redux";
import { getInitialData } from "./services";
import { Store } from "redux";

declare global {
  interface Window {
    reduxStore: Store;
    Stage: string;
  }
}

getInitialData()
  .then(data => {
    store.dispatch({
      type: "LOAD_INIT_DATA",
      payload: { ...data }
    });
    if (window.location.hash.includes("?token=")) {
      const tokenString = window.location.hash.split("?token=");
      window.localStorage.setItem("accessToken", tokenString[1]);
    }

    window.reduxStore = store;
    const Main = () => (
      <Provider store={store}>
        <App />
      </Provider>
    );
    ReactDOM.render(<Main />, document.getElementById("root"));
  })
  .catch(ex => {
    ReactDOM.render(
      <div>Error loading config</div>,
      document.getElementById("root")
    );
  });
