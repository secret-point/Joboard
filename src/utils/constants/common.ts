export const HVH_LOCALE = 'hvh-locale';
