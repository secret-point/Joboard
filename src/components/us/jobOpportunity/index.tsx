import React, { useEffect } from "react";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import JobOpportunity from "./JobOpportunity";
import JobConfirmation from "./JobConfirmation";
import { IconArrowLeft, IconSize } from "@amzn/stencil-react-components/icons";
import { Text } from "@amzn/stencil-react-components/text";
import { JOB_OPPORTUNITY_PAGE } from "../../../utils/enums/common";
import { CommonColors } from "../../../utils/colors";
import JobDescription from "./JobDescription";
import { translate as t } from "../../../utils/translator";
import { connect } from "react-redux";
import { uiState } from "../../../reducers/ui.reducer";
import { boundSetJobOpportunityPage } from "../../../actions/UiActions/boundUi";

interface MapStateToProps {
    ui: uiState
}

const JobOpportunityContainer = (props: MapStateToProps) => {
    const { ui } = props;
    const { jobOpportunity } = ui;
    const { currentPage } = jobOpportunity;
    const { INDEX, JOB_CONFIRMATION, JOB_DESCRIPTION } = JOB_OPPORTUNITY_PAGE;

    useEffect(() => {
        window.scroll(0, 0);
    })

    const renderPage = ( pageType: JOB_OPPORTUNITY_PAGE ) => {
        switch(pageType) {
            case INDEX:
                return <JobOpportunity/>;

            case JOB_CONFIRMATION:
                return <JobConfirmation />;

            case JOB_DESCRIPTION:
                return <JobDescription/>;

            default:
                return <JobOpportunity />
        }
    }


    const handleBack = () => {
        if(currentPage === JOB_CONFIRMATION) {
            boundSetJobOpportunityPage(INDEX);
        }
        else {
            boundSetJobOpportunityPage(JOB_CONFIRMATION);
        }
    }

    return (
        <Col>
            {
                currentPage !== INDEX &&
                <Row
                    gridGap={5}
                    alignItems="center"
                    width="fit-content"
                    color={CommonColors.Blue70}
                    padding='S200'
                    onClick={handleBack}
                    style={{ cursor: 'pointer' }}
                >
                    <IconArrowLeft size={IconSize.ExtraSmall}/>
                    <Text fontSize="T100" fontWeight="medium">
                        {currentPage === JOB_CONFIRMATION ?
                            t('BB-JobOpportunity-back-to-indexPage-link', 'View Jobs') :
                            t('BB-JobOpportunity-back-to-jobConfirmation-link','Back')
                        }
                    </Text>
                </Row>
            }
            {
                renderPage(currentPage)
            }
        </Col>
    )
}

const mapStateToProps = ( state: MapStateToProps ) => {
    return state;
};

export default connect(mapStateToProps)(JobOpportunityContainer)
