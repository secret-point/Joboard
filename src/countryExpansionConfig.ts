import { CountryCode, DESIRED_WORK_HOURS, SCHEDULE_FILTER_TYPE } from "./utils/enums/common";
import { ScheduleSortBy } from "./utils/types/common";

export const thankYouPageRedirectTextBanner = {
  US: { translationKey: "BB-ThankYou-fill-wotc-description-text", defaultString: "Before your pre-hire appointment, fill out Work Opportunities Tax Credit Questionnaire." },
  MX: { translationKey: "BB-ThankYou-redirect-to-ash-text", defaultString: "Before your pre-hire appointment, you will need to complete a series of pre-hire activities." },
  CA: { translationKey: "BB-ThankYou-fill-wotc-description-text", defaultString: "Before your pre-hire appointment, fill out Work Opportunities Tax Credit Questionnaire." }, // TODO: set to correct values once they're available
  UK: { translationKey: "BB-ThankYou-fill-wotc-description-text", defaultString: "Before your pre-hire appointment, fill out Work Opportunities Tax Credit Questionnaire." },
};

export enum PayRateType {
  hourMin = "hourMin",
  hourMax = "hourMax",
  monthMin = "monthMin",
  monthMax = "monthMax"
}

export interface countryConfigType {
  payRateType: PayRateType;
  desiredWorkHours: DESIRED_WORK_HOURS;
  nameRegexValidator: string;
  previousLegalNameRegexValidator: string;
  addressRegexValidator: string;
  renderScheduleCardBanner: (currencyCode: string, signOnBonus: number, signOnBonusL10N: string) => string;
}

export const accentedChars = "À-ÖØ-öø-ÿ"; // for Spanish
export const specialChars = `-!${accentedChars}`; // allow - and ! for the moment
export const digits = "0-9"; // same as \d
export const alphabet = "a-zA-Z";
export const alphanumeric = `${alphabet}${digits}`; // different with \w, no `_`

export const countryConfig: { [key in CountryCode]: countryConfigType } = {
  [CountryCode.US]: {
    payRateType: PayRateType.hourMax,
    desiredWorkHours: {
      RANGE1: "10",
      RANGE2: "20",
      RANGE3: "30",
      RANGE4: "40"
    },
    nameRegexValidator: `^(?=\\S)[${alphabet}${specialChars} ,.'-]{1,39}[${alphabet}${accentedChars}]$`,
    previousLegalNameRegexValidator: `^(?=\\S)[${alphabet}${specialChars},.'-]{1,19}\\s[${alphabet}${specialChars},.'-]{1,19}$`,
    addressRegexValidator: `^(?=\\S)[${alphanumeric}${specialChars} ]{1,}[${alphanumeric}${accentedChars}]$`,
    renderScheduleCardBanner: (currencyCode: string, signOnBonus: number, signOnBonusL10N: string) => {
      return signOnBonusL10N || `${currencyCode}${signOnBonus}`;
    },
  },
  [CountryCode.MX]: {
    payRateType: PayRateType.monthMax,
    desiredWorkHours: {
      RANGE0: "24",
      RANGE1: "30",
      RANGE2: "36",
      RANGE3: "42",
      RANGE4: "48"
    },
    nameRegexValidator: `^(?=\\S)[${alphabet}${specialChars} ,.'-]{1,99}[${alphabet}${accentedChars}]$`,
    previousLegalNameRegexValidator: `^(?=\\S)[${alphabet}${specialChars},.'-]{1,49}\\s[${alphabet}${specialChars},.'-]{1,49}$`,
    addressRegexValidator: `^(?=\\S)[${alphanumeric}${specialChars},. ]{1,}[${alphanumeric}${accentedChars},.]$`,
    renderScheduleCardBanner: (currencyCode: string, signOnBonus: number, signOnBonusL10N: string) => {
      return `${currencyCode}${signOnBonusL10N}`;
    },
  },
  // TODO: set to correct values once they're available
  [CountryCode.CA]: {
    payRateType: PayRateType.hourMax,
    desiredWorkHours: {
      RANGE1: "10",
      RANGE2: "20",
      RANGE3: "30",
      RANGE4: "40"
    },
    nameRegexValidator: `^(?=\\S)[${alphabet}${specialChars} ,.'-]{1,39}[${alphabet}${accentedChars}]$`,
    previousLegalNameRegexValidator: `^(?=\\S)[${alphabet}${specialChars},.'-]{1,19}\\s[${alphabet}${specialChars},.'-]{1,19}$`,
    addressRegexValidator: `^(?=\\S)[${alphanumeric}${specialChars} ]{1,}[${alphanumeric}${accentedChars}]$`,
    renderScheduleCardBanner: (currencyCode: string, signOnBonus: number, signOnBonusL10N: string) => {
      return signOnBonusL10N || `${currencyCode}${signOnBonus}`;
    },
  },
  [CountryCode.UK]: {
    payRateType: PayRateType.hourMax,
    desiredWorkHours: {
      RANGE1: "10",
      RANGE2: "20",
      RANGE3: "30",
      RANGE4: "40"
    },
    nameRegexValidator: `^(?=\\S)[${alphabet}${specialChars} ,.'-]{1,39}[${alphabet}${accentedChars}]$`,
    previousLegalNameRegexValidator: `^(?=\\S)[${alphabet}${specialChars},.'-]{1,19}\\s[${alphabet}${specialChars},.'-]{1,19}$`,
    addressRegexValidator: `^(?=\\S)[${alphanumeric}${specialChars} ]{1,}[${alphanumeric}${accentedChars}]$`,
    renderScheduleCardBanner: (currencyCode: string, signOnBonus: number, signOnBonusL10N: string) => {
      return signOnBonusL10N || `${currencyCode}${signOnBonus}`;
    },
  },
};

export const CS_DOMAIN_LIST = [
  // US
  "https://beta-us.devo.jobsatamazon.hvh.a2z.com",
  "https://gamma-us.devo.jobsatamazon.hvh.a2z.com",
  "https://us.preprod.jobsatamazon.hvh.a2z.com",
  "https://hiring.amazon.com",

  // MX
  "https://beta-mx.devo.jobsatamazon.hvh.a2z.com",
  "https://gamma-mx.devo.jobsatamazon.hvh.a2z.com",
  "https://mx.preprod.jobsatamazon.hvh.a2z.com",
  "https://trabaja.amazon.com.mx",

  // UK
  "https://beta.devo.jobsatamazon.hvh.a2z.com",
  "https://gamma.devo.jobsatamazon.hvh.a2z.com",
  "https://uk.preprod.jobsatamazon.hvh.a2z.com",
  "https://www.jobsatamazon.co.uk"
];

export const getCountryCode = (): CountryCode => {
  return "{{Country}}" as CountryCode || CountryCode.US;
};

export const getCountryConfig = (countryCode: CountryCode) => {
  return countryConfig[countryCode] || countryConfig[CountryCode.US];
};

export const getDesiredWorkHoursByCountryCode = (countryCodeOverride?: CountryCode) => {
  return getCountryConfig(countryCodeOverride || getCountryCode()).desiredWorkHours;
};

export const getScheduleSortList = () => {
  const ScheduleSortList: ScheduleSortBy[] = [
    {
      title: "Pay rate - Highest to Lowest",
      value: SCHEDULE_FILTER_TYPE.PAY_RATE,
      translationKey: "BB-JobOpportunity-sort-schedule-by-pay-rate-high-to-least"
    },
    {
      title: "Hours - Most to Least",
      value: SCHEDULE_FILTER_TYPE.HOURS_DESC,
      translationKey: "BB-JobOpportunity-sort-schedule-by-hours-most-to-least"
    },
    {
      title: "Hours - Least to Most",
      value: SCHEDULE_FILTER_TYPE.HOURS_ASC,
      translationKey: "BB-JobOpportunity-sort-schedule-by-hours-least-to-most"
    }
  ];

  switch (getCountryCode()) {
    case CountryCode.UK:
      ScheduleSortList.unshift({
        title: "Featured",
        value: SCHEDULE_FILTER_TYPE.FEATURED,
        translationKey: "BB-JobOpportunity-sort-schedule-by-featured"
      });
      break;
  
    default:
      break;
  }

  return ScheduleSortList;
};