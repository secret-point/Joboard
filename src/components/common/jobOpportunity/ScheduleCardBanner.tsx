import React from 'react';
import { Row } from "@amzn/stencil-react-components/layout";
import { getCountryCode, getCountryConfig } from '../../../countryExpansionConfig';
import { translate as t } from "../../../utils/translator";

interface ScheduleCardBannerProps {
  currencyCode: string;
  signOnBonus: number;
  signOnBonusL10N: string;
}

export const ScheduleCardBanner = (props: ScheduleCardBannerProps) => {
  const { currencyCode, signOnBonus, signOnBonusL10N } = props;

  const { renderScheduleCardBanner } = getCountryConfig(getCountryCode());

  return (
    <Row className="scheduleCardBanner">
      <div>
        {t("BB-Schedule-card-bonus-text", "Bonus:")} {renderScheduleCardBanner(currencyCode, signOnBonus, signOnBonusL10N)}
      </div>
    </Row>
  )
}

export default ScheduleCardBanner;
