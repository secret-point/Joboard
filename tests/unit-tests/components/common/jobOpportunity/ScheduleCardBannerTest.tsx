import React from "react";
import { shallow } from "enzyme";
import { ScheduleCardBanner } from "../../../../../src/components/common/jobOpportunity/ScheduleCardBanner";
import * as countryExpansionConfig from "../../../../../src/countryExpansionConfig";
import { CountryCode } from "../../../../../src/utils/enums/common";

const getCountryCodeSpy = jest.spyOn(countryExpansionConfig, "getCountryCode");

describe("ScheduleCardBanner", () => {
  describe("country code US", () => {
    beforeEach(() => {
      getCountryCodeSpy.mockReturnValue(CountryCode.US);
    });

    it("should match snapshot", () => {
      const shallowWrapper = shallow(
        <ScheduleCardBanner
          currencyCode="USD"
          signOnBonus={5.00}
          signOnBonusL10N="$5.00"
        />);

      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('country code MX', () => {
    beforeEach(() => {
      getCountryCodeSpy.mockReturnValue(CountryCode.MX);
    });

    it("should match snapshot", () => {
      const shallowWrapper = shallow(
        <ScheduleCardBanner
          currencyCode="MXN"
          signOnBonus={5.00}
          signOnBonusL10N="$5.00"
        />);

      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('country code CA', () => {
    beforeEach(() => {
      getCountryCodeSpy.mockReturnValue(CountryCode.CA);
    });

    it("should match snapshot", () => {
      const shallowWrapper = shallow(
        <ScheduleCardBanner
          currencyCode="CAD"
          signOnBonus={5.00}
          signOnBonusL10N="$5.00"
        />);

      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
