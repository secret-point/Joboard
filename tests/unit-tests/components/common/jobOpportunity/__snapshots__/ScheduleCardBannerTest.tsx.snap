// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleCardBanner country code CA should match snapshot 1`] = `
<Row
  className="scheduleCardBanner"
>
  <div>
    Bonus:
     
    $5.00
  </div>
</Row>
`;

exports[`ScheduleCardBanner country code MX should match snapshot 1`] = `
<Row
  className="scheduleCardBanner"
>
  <div>
    Bonus:
     
    MXN$5.00
  </div>
</Row>
`;

exports[`ScheduleCardBanner country code US should match snapshot 1`] = `
<Row
  className="scheduleCardBanner"
>
  <div>
    Bonus:
     
    $5.00
  </div>
</Row>
`;
